src/libs/signer.service.ts

```diff
@@ -25,34 +32,77 @@ export function getSigner() {
   return signer;
 }
 
-export async function signTransaction(signingRequest: SigningRequest) {
+export async function signTransaction(
+  signingRequest: SigningRequest
+): Promise<SigningResponse> {
   const signer = getSigner();
   const provider = signer.provider;
 
   if (!provider) throw new Error("RPC provider not specified");
 
+  let signerBalanceInWei: ethers.BigNumber;
+  try {
+    signerBalanceInWei = await signer.getBalance();
+  } catch (error) {
+    logger.error("❌ Unable to get signing key balance.");
+    throw error;
+  }
+
+  logger.info(
+    `💰 Signer address ${
+      signer?.address
+    } has a balance of ${ethers.utils.formatEther(signerBalanceInWei)} LYX`
+  );
+
   const { to, gasLimit, transactionData } = signingRequest;
 
+  if (Number(gasLimit) > Number(signerBalanceInWei)) {
+    const errorMessage = `😢 Insufficiant balance. Gas Limit : ${gasLimit} SignerKeyBalance : ${signerBalanceInWei}`;
```

Comment:
this message can be improved. "Signing key has insufficient balance to execute transaction" or something